plugins/org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/internal/AbstractTextSegment.java

```diff
@@ -61,20 +61,31 @@ public int getEndOffset() {
 	}
 
 	@Override
-	public ITextSegment getIndentation() {
-		return getTextRegionAccess().indentationRegion(getOffset());
+	public int getLineCount() {
+		return Strings.countLines(getText());
```

Comment:
I think you may want to use org.eclipse.xtext.util.Strings.countLineBreaks(CharSequence) instead